### ISSUE TITLE
Assertion failed: `it!=PAGNodeToActualParmMap.end() && "acutal parameter VFG node can not be found??"'

### ISSUE THREAD

**Issue by HolmesSherlock** [SVF-tools/SVF#153]
While performing a flow-sensitive analysis of the attached C program, SVF-1.7 (commit# 878c3b1eca06f569760ee312996b623986d3525d) fails with the following assertion:

```
$ wpa -fspta -print-pts svf_crash.bc

wpa: svf/include/Util/VFG.h:186: ActualParmVFGNode* VFG::getActualParmVFGNode(const PAGNode*, CallSite) const: Assertion `it!=PAGNodeToActualParmMap.end() && "acutal parameter VFG node can not be found??"' failed.
```

[svf_crash.zip](https://github.com/SVF-tools/SVF/files/3699920/svf_crash.zip)

**Comment by yuleisui** [comment]
Hi Dipanjan,

Thanks for the report. The issue has just been fixed a968c3f9369de9e63c591dd1cb0173b71f2d4eea.

The assertion is because SVF didn't check the types of both the actual and formal parameters of a vararg function.

In your case, the SVF treats all formal varargs as one vararg pag node using pointer type, while the actual parameter is an integer type. This scalar parameter passing should not be involved in pointer analysis.

**Comment by HolmesSherlock** [comment]
Thanks for the prompt fix and the explanation for the crash. I can confirm the fix works.
